jetty-quickstart/src/main/java/org/eclipse/jetty/quickstart/ExtraXmlDescriptorProcessor.java

```diff
@@ -68,11 +68,19 @@ public void end(WebAppContext context, Descriptor descriptor)
     {
     }
 
+    public void setOriginAttribute(String name)
+    {
+        _originAttribute = name;
+    }
+    
     public void saveSnippet(WebAppContext context, Descriptor descriptor, XmlParser.Node node)
         throws Exception
     {
+        //Note: we have to output the origin as a comment field instead of
+        //as an attribute like the other other elements because
+        //we are copying these elements _verbatim_ from the descriptor
         LOG.debug("save {}", node.getTag());
-        if (_showOrigin)
+        if (!StringUtil.isBlank(_originAttribute))
```

Comment:
Can we actually use the `_originAttribute` when precomputing like:
```java
origin = StringUtil.isBlank(_originAttribute)) ? null : ("  <!-- " +  _originAttribute + "=" + descriptor + " -->\n")
``` 
and then the code here can just be:
```suggestion
        if (_origin != null)
```